pep-0514.txt

```diff
@@ -118,17 +138,22 @@ It is not possible to detect side-by-side installations of both 64-bit and
 current user. Python 3.5 and later always uses different Tags for 64-bit and
 32-bit versions.
 
-Environments registered under other Company names must use distinct Tags to
-support side-by-side installations. Tools consuming these registrations are
-not required to disambiguate tags other than by preferring the user's setting.
+The following section describe user-visible information that may be registered.
+For Python 3.5 and earlier, none of this information is available, but
+alternative defaults are specified for the ``PythonCore`` key.
+
+Environments registered under other Company names have no backward compatibility
+requirements and must use distinct Tags to support side-by-side installations.
+Tools consuming these registrations are not required to disambiguate tags other
+than by preferring the user's setting.
 
 Company
 -------
 
 The Company part of the key is intended to group related environments and to
 ensure that Tags are namespaced appropriately. The key name should be
 alphanumeric without spaces and likely to be unique. For example, a trademarked
-name, a UUID, or a hostname would be appropriate::
+name (preferred), a UUID, or a hostname would be appropriate::
 
```

Comment:
For console users, a full file path (even one inside `%LocalAppData%`) is reasonable, but most Windows users don't have a clue where it's installed. I don't really expect this pep to add much value for virtualenv directly - the value really is for point-and-click types (but I also didn't want to exclude console users, hence I wasn't totally explicit about this).

I don't think malicious name squatting will be an issue, but accidental name squatting is already an issue. This PEP is meant to be guidance to avoid accidental collisions in the registry, so recommending unique and/or defensible names is more important than convenient user abbreviations.

I can see a use for having a hash of a repository URL and a commit ID as a Company/Tag pair in a distributed build system, for example. In this case, you'd almost certainly be passing around the values automatically.